moxy-compiler/src/main/java/moxy/compiler/Util.java

```diff
@@ -197,4 +203,29 @@ public static String decapitalizeString(String string) {
         return string == null || string.isEmpty() ? "" : string.length()
             == 1 ? string.toLowerCase() : Character.toLowerCase(string.charAt(0)) + string.substring(1);
     }
+
+    public static <T> boolean equalsBy(
+            Collection<T> first,
+            Collection<T> second,
+            BiFunction<T, T, Boolean> predicate) {
+
+        Preconditions.checkArgument(predicate != null, "Require non null predicate!");
+        if (first != null && second != null) {
+            if (first.size() != second.size()) {
+                return false;
+            }
+            final Iterator<T> firstIterator = first.iterator();
+            final Iterator<T> secondIterator = second.iterator();
+            while (firstIterator.hasNext()) {
+                final T firstItem = firstIterator.next();
+                final T secondItem = secondIterator.next();
+                if (!predicate.apply(firstItem, secondItem)) {
+                    return false;
+                }
+            }
+            return true;
+        } else {
+            return first != second;
```

Comment:
1)If A == null and B == null -> null != null -> true
2)If A != null and B == null  -> A != null -> true // But we want false?
3)If A != null and B == null  -> B != null -> true // But we want false?